visualizer/IdealGraphVisualizer/SourceRepository/src/main/java/org/graalvm/visualizer/source/impl/ui/LocationOpener.java

```diff
@@ -63,24 +67,44 @@ private void openOrView(boolean focus) {
             return;
         }
         int line = location.getLine();
+        int[] offsets = location.getOffsetsOrNull();
         EditorCookie cake = toOpen.getLookup().lookup(EditorCookie.class);
-
-        Line l;
-        try {
-            l = cake.getLineSet().getOriginal(line - 1);
-        } catch (IndexOutOfBoundsException ex) {
-            // expected, the source has changed
-            return;
+        if (cake.getDocument() != null && offsets != null) {
+            line = NbDocument.findLineNumber(cake.getDocument(), offsets[0]);
         }
+
+        Task task = cake.prepareDocument();
+
+        TaskListener[] whenShowing = new TaskListener[1];
+        whenShowing[0] = (ev) -> {
+            task.removeTaskListener(whenShowing[0]);
+            Mutex.EVENT.postReadRequest(() -> {
+                if (offsets == null) {
+                    StatusDisplayer.getDefault().setStatusText(Bundle.ERR_LineNotFound());
```

Comment:
I'm not sure if I'm understanding the control flow here but it looks to me that if the location had a valid line but no offsets you'd report `ERR_LineNotFound`.  Am I missing some subtlety?

Reply:
- I think you are right Tom, 8955e696540584b49748dd247290bee1b0aa892a would address that
- I have to admit I am not sure if `JEditorPane.select` is the right way to highlight the selection
- maybe there is something better like attaching some annotation to a [Line.Part](https://bits.netbeans.org/dev/javadoc/org-openide-text/org/openide/text/Line.Part.html)
- I can try to implement whatever you guys think is the best way to handle the highlight

Reply:
I cannot find the info explicitly spelled in the [javadoc of Truffle Source](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/source/Source.html), but I believe a _newline_ separator is always treated as a single character when counting _offset_ in the source.

NetBeans does treat _newline_ as a single character as well.

As such I believe this protocol shall do the same: count _newline_ as a single character when it comes to offsets.